bundle/src/main/java/com/adobe/acs/tools/explain_query/impl/ExplainQueryServlet.java

```diff
@@ -380,21 +423,23 @@ private JSONArray compositeQueryDataToJSON(Collection<CompositeData> queries) th
         return jsonArray;
     }
 
-    private String startCollection(){
+    private String startCollection() {
         final String collectorKey = UUID.randomUUID().toString();
         MDC.put(QueryLogCollector.COLLECTOR_KEY, collectorKey);
         return collectorKey;
     }
 
-    private void stopCollection(String key){
-        MDC.remove(key);
-        if (logCollector != null){
-            logCollector.stopCollection(key);
+    private void stopCollection(String key) {
+        synchronized (this.logCollector) {
```

Comment:
Made synchronized since all executions of this servlet share the same QueryLogCollector